--- /home/ubuntu/OutputPage.php	2011-05-31 22:39:52.821880494 +0000
+++ OutputPage.php	2011-05-31 10:55:41.117410821 +0000
@@ -7,6 +7,7 @@
  */
 class OutputPage {
 	var $mMetatags = array(), $mKeywords = array(), $mLinktags = array();
+        var $mLinkCopyright = false;
 	var $mExtStyles = array();
 	var $mPagetitle = '', $mBodytext = '', $mDebugtext = '';
 	var $mHTMLtitle = '', $mIsarticle = true, $mPrintable = false;
@@ -141,7 +142,13 @@
 
 	function addLink( $linkarr ) {
 		# $linkarr should be an associative array of attributes. We'll escape on output.
-		array_push( $this->mLinktags, $linkarr );
+                if ( !array_key_exists('rel', $linkarr) || ($linkarr['rel'] <> 'copyright') || !$this->mLinkCopyright) {
+                  array_push( $this->mLinktags, $linkarr );
+                }
+                if (array_key_exists('rel', $linkarr) && ($linkarr['rel'] == 'copyright')) {
+                  $this->mLinkCopyright = true;
+                }
+
 	}
 	
 	# Get all links added by extensions
